### PR TITLE
Added code to store a select number of special characters and not htm…

### PR DESCRIPTION
**Pull request by jbradl11**
Pull requests into Bonnie Bundler require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

This PR fixes this issue of supporting special characters in CQL define statements. 

Example: 
```
define: '1 & 2' :
  stuff
```
becomes
```
define: '1 &amp; 2':
  stuff
```
This PR will keep the "special characters" if they are one of the following: `&`, `"`, `<`, `>`, and `'`

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1583
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here:
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] ~JIRA test links:~
- [x] ~Justification for using JIRA tests:~
- [x] ~JIRA tests have been added to sprint~

**Reviewer 1:**

Name: @mayerm94
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: lib/measures/elm_parser.rb [hunk 1]
```diff
@@ -47,9 +48,14 @@ def self.parse_node(node)
         else
           if (/^define/ =~ child.to_html)
             define_name = child.to_html.split("\"")[1]
+            # Modify special characters back in the the define_name
+            @html_hash.each { |k,v| define_name.gsub!(k.to_s, v) }
           end
+          clause_text = child.to_html.gsub(/\t/, "  ")
```
Comment (mayerm94) [inline lib/measures/elm_parser.rb:54]: Can the tab to spaces mapping be added to your `@html_hash`. I think '@html_hash` would have to be renamed to `@character_substitutions` or something along those lines in that case

Comment (jbradl11) [reply lib/measures/elm_parser.rb:54]: Thats an interesting idea, I'll give that a shot.

Comment (jbradl11) [reply lib/measures/elm_parser.rb:54]: After looking closer at the code, I am not sure if this is a great change to make right now. The tab replacement only happens for the clause_text and not the define_name. I am not confident in making this change without knowing the full consequences. @mayerm94 thoughts?

Comment (mayerm94) [reply lib/measures/elm_parser.rb:54]: I can agree with that -- Given the unknown consequences, lack of testing on the view and the time crunch

**Review by mayerm94 — APPROVED** [review]

**Review by hossenlopp — APPROVED** [review]
